app/src/main/java/zapsolutions/zap/setup/ConnectRemoteNodeActivity.java

```diff
@@ -107,75 +116,108 @@ public void onClick(View v) {
-                            validParams = false;
-                        }
                     }
 
-                    if (validParams) {
-                        // Everything is ok, initiate connection
-                        connect(connectURI.getHost(), connectURI.getPort(), cert, macaroon);
-                    } else {
-                        ZapLog.debug(LOG_TAG, "Connect URI has invalid parameters (certificate or macaroon)");
-                        showError(getResources().getString(R.string.error_invalidRemoteConnectionString), 4000);
+                    try {
+                        BaseEncoding.base64Url().decode(macaroon);
+                    } catch (IllegalArgumentException e) {
+                        validParams = false;
                     }
+                }
 
+                if (validParams) {
+                    // Everything is ok, initiate connection
+                    connect(connectURI.getHost(), connectURI.getPort(), cert, macaroon);
                 } else {
-                    ZapLog.debug(LOG_TAG, "Connect URI has no parameters");
-                    showError(getResources().getString(R.string.error_invalidRemoteConnectionString), 4000);
+                    throw new IllegalArgumentException("Connect URI has invalid parameters (certificate or macaroon)");
                 }
+            } else {
+                throw new IllegalArgumentException("Connect URI has no parameters");
             }
         } catch (URISyntaxException e) {
             ZapLog.debug(LOG_TAG, "URI could not be parsed");
-            e.printStackTrace();
+            showError(getResources().getString(R.string.error_invalidRemoteConnectionString), 4000);
+        } catch (IllegalArgumentException e) {
+            ZapLog.debug(LOG_TAG, e.getMessage());
             showError(getResources().getString(R.string.error_invalidRemoteConnectionString), 4000);
         }
+    }
 
+    private void connectToBtcPay(String btcPayConfigurationJson) {
+        try {
+            // parse config
+            BTCPayConfigurationJson btcPayConfigurations = new Gson().fromJson(btcPayConfigurationJson, BTCPayConfigurationJson.class);
+            BTCPayConfiguration btcPayConfiguration = btcPayConfigurations.getConfiguration("grpc", "BTC");
+
+            // connect
+            if (btcPayConfiguration != null) {
+                mUG = new UserGuardian(this, DialogName -> {
+                    if (UserGuardian.BTC_PAY_CONNECT.equals(DialogName)) {
+                        connect(btcPayConfiguration.getHost(), btcPayConfiguration.getPort(), null, btcPayConfiguration.getMacaroon());
+                    }
+                });
+                mUG.securityConnectToBtcPay(btcPayConfiguration.getHost());
+            } else {
+                ZapLog.debug(LOG_TAG, "BTCPay Configuration does not contain BTC gRPC config");
+                showError(getResources().getString(R.string.error_invalidRemoteConnectionString), 4000);
```

Comment:
Can you please make a descriptive Error that is shown to the end user?
Something like: "Only BTCPay configurations for BTC with gRPC are supported."

Reply:
Yes, will update after https://github.com/LN-Zap/zap-android/pull/45 got merged 👍